doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   key - required argument, a string of what tag key to return values for
+// Returns: []string
+select(database:"foo").values(key:"metric")
+```
+
+#### Cardinality
+`cardinality` returns a map of the count of unique tag values for each key in the matrix.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   keys - optional string array. only return the cardinality for the passed keys
+// Returns: matrix
+select(database:"foo", where:{'region'="west"})
+  .cardinality()
+```
+
+#### Limit
+`limit` accepts either a matrix or an array and will limit the number of results. In the case of a matrix, it limits the number of series (or rows) that are returned. In the case of an array it is used to take a slice. It's a good practice to always apply a limit of some kind.
+
+```javascript
+// Accepts: matrix, array
+// Arguments:
+//   n - the number of results to limit to, required argument.
+//   offset - the offset (inclusive) to start with. defaults to 0.
+// Returns: a matrix if passed a matrix, an array if passed an array
+//
+select(database:"foo").limit(n:10,offset:20) // get the third page of results
+```
+
+#### Time Shift
+`timeShift` will shift all returned timestamps in the matrix by the given duration.
+
+```javascript
+// Accepts: a matrix
+// Arguments:
+//   duration - a duration
+// Returns: matrix
+select(database:"foo").range(start:"2017-01-01")
+  .timeShift(duration:-168h) // shift everything to a week before
+```
+
+#### Interpolate
+`interpolate` will normalize the time series in a matrix based on different criteria. It could work across columns or on the series themselves.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   start - time, ensure data exists starting at this time
+//   every - duration, ensure that a value exists at every duration period from start
+//   value - integer, float, string, bool, or expression to fill with. Expressions can
+//           use functions calculated from the series like mean($)
+// Returns: a matrix
+select(database:"foo",where:{'metric'="requests"})
+  .range(start:-4h)
+  .clear()
+  .window(period:10m)
+  .sum()
+  .interpolate(start:-4h,every:10m,value:{mean($)})
+```
+
+#### Join
+`join` operates on the columns in a matrix to combine series together. The `interpolate` function can be used to ensure that timestamps for the series line up and can be matched together.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   keys - the tag keys to group the join by
+//   exp - the expression to join series on
+// Returns: matrix
+select(database:"foo",where:{'metric'="errors" OR 'metric'="requests"})
+  .range(start:-30m)
+  .window(period:10m)
+  .sum()
+  .interpolate(start:-30m,value:0)
+  .join(keys:["host"],exp:
+    {('metric'."errors".float()/'metric'."requests" * 100).float(fixed:1) AS 'metric'."error_rate"})
```

Comment:
What happens if errors = 10 and requests = 0? 10/0? `null`, `0.0`, `none`, `""`?